File: 평강/app/project/src/main/java/com/eventpurchase/project/global/common/response/ErrorResponse.java
package com.eventpurchase.project.global.common.response;

import com.eventpurchase.project.global.common.exception.ErrorCode;

public record ErrorResponse (
        int status,
        String message



)
{
    public static ErrorResponse of(ErrorCode errorcode) {
        return new ErrorResponse(errorcode.getStatus().value(), errorcode.getMessage());
    }


}

File: 평강/app/project/shared/src/main/kotlin/com/eventpurchase/project/shared/response/ErrorResponse.kt
package com.eventpurchase.project.shared.response

import com.eventpurchase.project.shared.exception.ErrorCode

data class ErrorResponse(
    val status: Int,
    val message: String
) {
    companion object {
        fun of(errorCode: ErrorCode) = ErrorResponse(errorCode.status, errorCode.message)
    }
}
